core/src/main/java/org/wso2/carbon/kernel/internal/DataHolder.java

```diff
@@ -81,4 +84,22 @@ public void setRuntimeManager(RuntimeManager runtimeManager) {
     public void setOptConfigResolver(Optional<ConfigResolver> optConfigResolver) {
         this.optConfigResolver = optConfigResolver;
     }
+
+    /**
+     * Getter method of ${@link ConfigProvider}.
+     *
+     * @return optConfigProvider
+     */
+    public Optional<ConfigProvider> getOptConfigProvider() {
```

Comment:
The naming convention is not correct. The return can be an optional but that does not need to be part of the method name.